helix-term/src/commands/typed.rs

```diff
@@ -732,6 +734,9 @@ pub fn write_all_impl(
             None
         };
 
+        let view = view_mut!(cx.editor);
+        doc.append_changes_to_history(view);
```

Comment:
I think we should solve this more generically and call `append_changes_to_history` and `ensure_cursor_in_view` before and after executing any typable command, i.e. in this block: https://github.com/helix-editor/helix/blob/64326698225999016300423c4b90f8a8c7b8f38b/helix-term/src/commands.rs#L197-L206

Few typable commands modify the buffer so both `append_changes_to_history`s would be a no-op in most cases. This would also prevent bugs like https://github.com/helix-editor/helix/issues/9619, https://github.com/helix-editor/helix/issues/5733, also see https://github.com/helix-editor/helix/pull/5790

Reply:
Actually this is not totally correct :point_up:

We should save checkpoints before/after each command in sequences too, in this block instead: https://github.com/helix-editor/helix/blob/b7b6f300841bb61d8833fee1c58d0e3670849b61/helix-term/src/ui/editor.rs#L836-L859

This doesn't solve all of the problems of writing in command sequences (like https://github.com/helix-editor/helix/issues/9705) but it should make history commits more consistent

Reply:
Oh fantastic, I was searching for something like this when writing this patch but couldn't find it. That makes for much nicer behavior.